src/lib.rs

```diff
@@ -827,23 +835,93 @@ impl SyncClient {
         }
     }
 
+    async fn request_wildcard_handler(
+        wrapped_client: Arc<Mutex<Self>>,
+        mut subscription_rx: MpscReceiver<Msg>,
+    ) {
+        let disconnecting = Self::disconnecting(Arc::clone(&wrapped_client));
+        pin_mut!(disconnecting);
+        loop {
+            // Select between the next message and disconnecting
+            let msg = match future::select(subscription_rx.next(), disconnecting).await {
+                Either::Left((Some(msg), unresolved_disconnecting)) => {
+                    disconnecting = unresolved_disconnecting;
+                    msg
+                }
+                Either::Left((None, _)) => break,
+                Either::Right(((), _)) => break,
+            };
+
+            let mut client = wrapped_client.lock().await;
+            let response_inbox = msg.subject().to_string();
+            if let Some(mut requester_tx) = client.request_inbox_mapping.remove(&response_inbox) {
+                requester_tx.send(msg).await.unwrap_or_else(|err| {
+                    warn!("Could not write response to pending request via mapping channel. Skipping! Err: {}", err);
+                });
+            }
```

Comment:
Could you add an `else` case that logs an error and has a `debug_assert!(false)`

Reply:
Based on the [nats.go client](https://github.com/nats-io/nats.go/blob/6eb6122d3827c87e99b9a095ef3667d5aedf3b37/nats.go#L2594-L2601) I don't think we can safely assert the `else` case won't/ shouldn't be hit. In golang, the zero-value nil channel is safe to non-blocking write to (fails silently) so if some bad-actor publishes a message to our wildcard subscription, the assert would panic, where the go client would continue just fine. Of course, the `debug_assert!` won't panic in release mode, but it's certainly expected behavior. Still happy to add it though. Just thought I'd point that out first.

Reply:
Ah, that is a good point. I agree; an assert does not make much sense, but I think a warn level message describing what is going on would be good to have.